packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantAvailability/RestaurantAvailability.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+    <div :class="[$style['c-restaurantCard-availability']]">
+        <icon-text
+            v-if="availabilityTranslatedName"
+            :text="availabilityTranslatedName"
+            :color="iconColor"
+            hide-icon-in-tile-view
+            data-test-id="restaurant-availability-type"
+            :class="[$style['c-restaurantCard-availability-iconText']]">
+            <component
+                :is="icon"
+                v-if="icon"
+                aria-hidden="true"
+                data-test-id="restaurant-availability-icon" />
+        </icon-text>
+
+        <p
+            v-if="detailsMessage"
+            data-test-id="restaurant-availability-detailsMessage"
+            :class="[$style['c-restaurantCard-availability-detailsMessage']]">
+            {{ detailsMessage }}
+        </p>
+    </div>
+</template>
+
+<script>
+import { ClockSmallIcon, WalkingSmallIcon } from '@justeat/f-vue-icons';
+import isValidAvailabilityType from './propValidators/availabilityType.validator';
+import availabilityTypes from './availabilityTypes';
+import IconText from '../IconText.vue';
+
+const availabilityTypeIcons = {
+    [availabilityTypes.PREORDER]: ClockSmallIcon,
+    [availabilityTypes.COLLECTION]: WalkingSmallIcon
+};
+
+const iconColors = {
+    [availabilityTypes.PREORDER]: 'blue'
+};
+
+export default {
+    name: 'RestaurantAvailability',
+    components: {
+        IconText,
+        ClockSmallIcon,
+        WalkingSmallIcon
+    },
+    props: {
+        // A string representation of pre-defined availability types: PREORDER, COLLECTION
+        availabilityType: {
+            type: String,
+            default: null,
+            validator: isValidAvailabilityType
+        },
+        // The translated version of the availability type for the current tenant i.e. 'Pre-order', 'Preordina'
+        availabilityTranslatedName: {
+            type: String,
+            default: null
+        },
+        // The accompanying message for the availability type. i.e Opening times, delivering from...
+        detailsMessage: {
+            type: String,
```

Comment:
No need to change this but `availabilityMessage` unifies it with the other props :)

Reply:
updated for consistency!